File: apps/motoboy/app/src/main/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/RouteStopDisplay.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import com.drogsantoantonio.farmadelivery.data.models.RouteStopDto

/**
 * Define o titulo principal da parada na rota.
 * Usa o codigo da entrega quando existir; sem entrega vinculada, mostra o tipo da parada.
 */
fun routeStopTitle(stop: RouteStopDto): String {
  val delivery = stop.delivery ?: return routeStopTypeLabel(stop.type)
  return deliveryPrimaryCodeLabel(delivery.publicCode, delivery.storeDailyNumber)
}

/**
 * Monta o detalhe da parada com cliente e status da entrega.
 * Retorna null quando nao ha entrega para nao poluir a tela com texto vazio.
 */
fun routeStopDetails(stop: RouteStopDto): String? {
  val delivery = stop.delivery ?: return null
  val customer = delivery.customer.trim()
  val status = deliveryStatusLabel(delivery.status)
  val code = if (deliveryDailyNumberLabel(delivery.storeDailyNumber) != null) delivery.publicCode else null
  val parts = listOfNotNull(code, customer.takeIf { it.isNotEmpty() }, status)

  return parts.joinToString(" - ")
}

File: apps/motoboy-pwa/src/routeStopDisplay.ts
import { deliveryDailyNumberLabel, deliveryPrimaryCodeLabel } from "./deliveryDisplayCode";
import { deliveryStatusLabel } from "./deliveryStatusLabels";
import { routeStopTypeLabel } from "./routeStopTypeLabels";
import type { CourierRoute } from "./types";

type RouteStop = CourierRoute["stops"][number];

/**
 * Define o titulo principal da parada na rota do motoboy.
 * Quando a API manda a entrega vinculada, mostramos o codigo publico; senao usamos o tipo da parada.
 */
export function routeStopTitle(stop: RouteStop): string {
  return stop.delivery ? deliveryPrimaryCodeLabel(stop.delivery) : routeStopTypeLabel(stop.type);
}

/**
 * Monta a linha complementar da parada com cliente e status.
 * Se a parada nao tiver entrega vinculada, nao forca texto extra.
 */
export function routeStopDetails(stop: RouteStop): string | null {
  if (!stop.delivery) return null;

  const customer = stop.delivery.customer.trim();
  const status = deliveryStatusLabel(stop.delivery.status);
  const dailyNumber = deliveryDailyNumberLabel(stop.delivery);
  const code = dailyNumber ? stop.delivery.publicCode : null;
  const parts = [code, customer, status].filter(Boolean);

  return parts.join(" - ");
}
